docs/install/build_from_source.md

```diff
@@ -28,184 +32,102 @@ You need C++ build tools and BLAS library to build MXNet shared library. If you
-   tar -zxf gcc-4.8.5.tar.gz
-   cd gcc-4.8.5
-   ./contrib/download_prerequisites
-   ```
+MXNet relies on the
+[BLAS](https://en.wikipedia.org/wiki/Basic_Linear_Algebra_Subprograms) (Basic
+Linear Algebra Subprograms) library for numerical computations. You can install
+any one among [ATLAS](http://math-atlas.sourceforge.net/),
+[OpenBLAS](http://www.openblas.net/) and
+[MKL](https://software.intel.com/en-us/intel-mkl).
 
-3. Build `gcc` by using 10 threads and then install to `/usr/local`
 
-   ```bash
-   mkdir release && cd release
-   ../configure --prefix=/usr/local --enable-languages=c,c++
-   make -j10
-   sudo make install
-   ```
+#### Optional
 
-4. Add the lib path to your configure file such as `~/.bashrc`:
+* [OpenCV](http://opencv.org/) for Image Loading and Augmentation
+* [NVDIA CUDA and cuDNN](https://developer.nvidia.com/cuda-downloads) for running MXNet with GPUs
 
-   ```bash
-   export LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:/usr/local/lib64
-   ```
 
-</div>
-</div> <!-- linux -->
+### macOS
 
-<div class="windows">
+Refer to the [MXNet macOS setup guide](osx_setup.html) for detailed instructions.
 
-1. If [Microsoft Visual Studio 2015](https://www.visualstudio.com/downloads/) is not already installed, download and install it. You can download and install the free community edition.
-2. Download and Install [CMake](https://cmake.org/) if it is not already installed.
 
-</div>
+### Windows
 
-<div class="macos">
+Refer to the [MXNet Windows setup guide](windows_setup.html) for detailed instructions.
 
-Install [Xcode](https://developer.apple.com/xcode/).
 
-</div>
+### Ubuntu
 
-#### BLAS library
+Refer to the <a href="ubuntu_setup.html">MXNet Ubuntu installation guide</a> for build from source instructions as well as installation of language bindings.
 
-MXNet relies on the
-[BLAS](https://en.wikipedia.org/wiki/Basic_Linear_Algebra_Subprograms) (Basic
-Linear Algebra Subprograms) library for numerical computations. You can install
-any one among [ATLAS](http://math-atlas.sourceforge.net/),
-[OpenBLAS](http://www.openblas.net/) and
-[MKL](https://software.intel.com/en-us/intel-mkl).
 
-<div class="linux">
-<div class="ubuntu">
+### CentOS
+1. Install build tools and git on `CentOS >= 7` and `Fedora >= 19`:
 
 ```bash
-sudo apt-get install libatlas-base-dev
+sudo yum groupinstall -y "Development Tools" && sudo yum install -y git
 ```
 
-</div>
-
-<div class="centos">
+2. Install Atlas:
 
 ```bash
 sudo yum install atlas-devel
 ```
 
-</div>
-
-<div class="linux">
-
-You can follow this link to build
-[OpenBlas from source](https://github.com/xianyi/OpenBLAS#installation-from-source).
-
-</div>
-</div>
-
-<div class="macos">
-
-macOS users can skip this step as `xcode` ships with a BLAS library.
-
-</div>
+### Other Linux
+Installing both `git` and `make` by following instructions on the websites is
+straightforward. Here we provide the instructions to build `gcc-4.8` from source codes.
```

Comment:
Are we sure that we need instructions how to build gcc? Is there any use case where it can not be installed with a package manager?

Say: `sudo apt install gcc-4.8` or `sudo apt install gcc-4.9`

Reply:
So, with Caffe2 on CentOS, I had to build gcc from scratch. I haven't tried MXNet on CentOS yet, but I wonder if whoever wrote this ran into a similar issue.

Since this is for "other linux" the `apt` instructions won't cut it. 
My preference would be to split out a separate page for CentOS and have thorough, tested instructions. (maybe as a separate PR?)